embedded/store/immustore.go

```diff
@@ -924,36 +847,44 @@ func (s *ImmuStore) appendData(entries []*EntrySpec, donec chan<- appendableResu
 	donec <- appendableResult{offsets, nil}
 }
 
-func (s *ImmuStore) Commit(txSpec *TxSpec) (*TxHeader, error) {
-	return s.commitUsing(txSpec, nil)
+func (s *ImmuStore) NewWriteOnlyTx() (*OngoingTx, error) {
+	return newWriteOnlyTx(s)
 }
 
-func (s *ImmuStore) commitUsing(txSpec *TxSpec, hdr *TxHeader) (*TxHeader, error) {
+func (s *ImmuStore) NewTx() (*OngoingTx, error) {
+	return newReadWriteTx(s)
+}
+
+func (s *ImmuStore) commit(otx *OngoingTx, expectedHeader *TxHeader, waitForIndexing bool) (*TxHeader, error) {
 	s.mutex.Lock()
 	if s.closed {
 		s.mutex.Unlock()
 		return nil, ErrAlreadyClosed
 	}
 	s.mutex.Unlock()
 
-	if txSpec == nil {
+	if otx == nil {
 		return nil, ErrIllegalArguments
 	}
 
-	err := s.validateEntries(txSpec.Entries)
+	err := s.validateEntries(otx.entries)
 	if err != nil {
 		return nil, err
 	}
 
+	if !otx.IsWriteOnly() && otx.snap.Ts() <= s.committedTxID {
```

Comment:
Can we safely read `s.committedTxID` without a lock ?

Reply:
no it's not